### PR TITLE
add codeql

### PR DESCRIPTION
**Pull request by Liaojinghui**
https://codeql.github.com/

### CODE REVIEW

**Comment by erikzhang** [comment]
Can you explain more about CodeQL?

**Comment by steven1227** [comment]
> Can you explain more about CodeQL?

A code analysis tool provided by Github ? . See [this](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-with-codeql)

**Comment by Liaojinghui** [comment]
> Can you explain more about CodeQL?

CodeQL is the code analysis engine developed by GitHub to automate security checks. You can analyze your code using CodeQL and display the results as code scanning alerts.

There are three main ways to use CodeQL analysis for code scanning:

- Use default setup to automatically configure CodeQL analysis for code scanning on your repository. The default setup chooses the languages to analyze, query suites to run, and events that trigger scans, then displays a summary of the analysis settings. After you enable CodeQL, GitHub Actions will execute workflow runs to scan your code. For more information, see "[Configuring code scanning for a repository](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning-for-a-repository#configuring-code-scanning-automatically)."

- Use advanced setup to add the CodeQL workflow to your repository. This generates a customizable workflow file which uses the [github/codeql-action](https://github.com/github/codeql-action/) to run the CodeQL CLI. For more information, see "[Configuring code scanning for a repository](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning-for-a-repository#creating-an-advanced-setup)."

- Run the CodeQL CLI directly in an external CI system and upload the results to GitHub. For more information, see "[About CodeQL code scanning in your CI system](https://docs.github.com/en/code-security/code-scanning/using-codeql-code-scanning-with-your-existing-ci-system/about-codeql-code-scanning-in-your-ci-system)."

**Comment by vncoelho** [comment]
@Liaojinghui, I checked the logs and asked re-run the job.
However, I did not get any analysis from it yet, what is the expected flow?
Perhaps we will need to create specify flows for analysis, right?

**Comment by Liaojinghui** [comment]
> @Liaojinghui, I checked the logs and asked re-run the job. However, I did not get any analysis from it yet, what is the expected flow? Perhaps we will need to create specify flows for analysis, right?

For most projects, this file is standard and by default, no change or configuration is required.

**Review by shargon — APPROVED** [review]

**Review by vncoelho — APPROVED** [review]
@Liaojinghui, I was searching for the report. That then I saw that it becomes a comment here made by the github bot.

Just checked it right now.

**Comment by vncoelho** [comment]
> You have successfully added a new CodeQL configuration `/language:csharp`. As part of the setup process, we have scanned this repository and found 2 existing alerts. Please check the repository [Security tab](/neo-project/neo/security/code-scanning?query=pr%3A2856+tool%3ACodeQL+is%3Aopen) to see all alerts.

For the currently reported I think it is OK because ECB is not encrypted twice.

**Comment by Liaojinghui** [comment]
> 

@vncoelho funny part, i can not read the report~ for security reason.

**Comment by vncoelho** [comment]
ahueaheuaea, but currently the report is in our favor.
None for neo-node and here it is just generic for ECB encryption.

"ECB should not be used as a mode for encryption. It has dangerous weaknesses. Data is encrypted the same way every time meaning the same plaintext input will always produce the same ciphertext. This makes encrypted messages vulnerable to replay attacks."
For the wallet `src/Neo/Wallets/Wallet.cs` and KeyPair.

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  schedule:
+    - cron: '34 22 * * 5'
```
Comment (vncoelho) [inline .github/workflows/codeql.yml:21]: What is this schedule? Every 5th at 22:34?

Comment (Liaojinghui) [reply .github/workflows/codeql.yml:21]: In the given cron schedule '34 22 * * 5', it means the CodeQL workflow will be triggered at:

34 minutes past the hour
22nd hour (10 PM)
Every day of the month
Every month
On the 5th day of the week (Friday)
So, the workflow will run every Friday at 10:34 PM.

Well, codeql has that, i did not change anything basically

Comment (vncoelho) [reply .github/workflows/codeql.yml:21]: ok

Comment (Liaojinghui) [reply .github/workflows/codeql.yml:21]: It did not mention time zone, then UTC maybe

**Comment by Liaojinghui** [comment]
@dotnet-policy-service agree [company="neo"]

**Pull request closed without merging**